Add optional breadcrumb to HeroImage

chrisrhymes opened the pull request:
Added the option to display the parent breadcrumb in the hero image. By default it is disabled. If set to display, it will display the breadcrumb above the title in the hero image. 

As part of this it was requested to make the hero image content the maximum width of the page. This involved more refactoring than I thought and ended up using flexbox for positioning and size, instead of position: absolute. Quite a big change for a small change. 

## To test
- `npm run test HeroImage`
- Check the HeroImage in structure, ensuring it displays as expected in different screen resolutions
- Check the HeroImage in the ServiceLandingPage examples, ensuring it displays as expected in different screen resolutions

chrisrhymes commented on src/library/structure/HeroImage/HeroImage.styles.js:
```diff
@@ -1,27 +1,44 @@
+  flex-direction: row;
+  align-items: flex-end;
+
+  &::before {
```
Use psudo element to control the aspect ratio. This should allow the height to grow with more content https://css-tricks.com/aspect-ratio-boxes/#aa-the-pseudo-element-tactic

chrisrhymes commented on src/library/structure/HeroImage/HeroImage.styles.js:
```diff
@@ -107,12 +120,21 @@ export const CallToActionLink = styled.a`
   }
 `;
+
+export const BreadcrumbLink = styled.a`
```
Added the breadcrumb link styles

DamienBurkeWN commented on src/library/structure/HeroImage/HeroImage.types.ts:
```diff
@@ -38,4 +40,14 @@ export interface HeroImageProps {
+  /**
+   * Should the breadcrumb be shown
+   */
+  showBreadcrumb?: boolean;
```
Do we need this? Could we instead rely on whether breadcrumbs array is empty or not?

DamienBurkeWN commented on src/library/structure/HeroImage/HeroImage.types.ts:
```diff
@@ -38,4 +40,14 @@ export interface HeroImageProps {
+  /**
+   * An optional array of breadcrumbs
+   */
+  breadcrumbsArray?: BreadcrumbProp[];
```
I know it has been done already in some other code but I don't think including the _type_ in the prop name is stylistically good so just **breadcrumbs** would be better. However, in this case, as we only ever look at the last one in the array, why do we even need it to be an array? I feel this component would be better off accepting a single **BreadcrumbProp** called **breadcrumb** (and being set or not handles the job of showBreadcrumb too).

DamienBurkeWN replied:
No, I'd do that in the frontend, we've got the breadcrumb trail at the page component level, easy enough to pass the last one through to this component. I think we can safely assume it'll never now be used as a slice/embedded para (and we can remove the support for that use in the frontend too).

DamienBurkeWN replied:
(if it was ever to be used as a slice, it surely wouldn't have a breadcrumb on it?)

DamienBurkeWN reviewed (changes requested):
I'm unclear on the 'make the hero image content the maximum width of the page' aspect - if that means moving the box/title area to align with the page width container, I'm sure there was a discussion about doing that and we agreed not to. The reasoning was that the hero image itself is chosen so that the bottom left is the 'safe' area to obscure with any content. If the box/title moves depending on the browser size, there is no 'safe' area anymore?

Anyway, if that is how it will be, there's some other issues now - on mobile screen sizes, the boxed content area is now no longer in the center (right side is closer to the scrollbar than before) and the call to action button loses its spacing from the bottom of the box.

As it is currently: 
<img width="328" alt="image" src="https://user-images.githubusercontent.com/91884003/158210264-0bfa9fbf-7876-422f-b040-8ac3150fbce7.png">

Compare now:
<img width="328" alt="image" src="https://user-images.githubusercontent.com/91884003/158210481-d7d9cb7f-67d0-4bb6-bbe3-2c69ac3c1e0b.png">

chrisrhymes commented:
Thanks for the feedback. The safe area is quite difficult to define, as the image ratio changes on mobile and desktop so you will see different parts of the image. The overlay box provides one solution and there is the linear gradient when the box is not used. I think whatever we do it will need a bit of trial and error by the content editors to work out what works best. 

I have tidied up the mobile and tablet styles so it looks more like before. Please can you retest. 

I have removed the showBreadcrumb prop and updated the `breadcrumbsArray` to be a single `breadcrumb`. If it's not provided then it will not be displayed.

DamienBurkeWN reviewed (approved):
Nice 👍

chrisrhymes commented:
Good spot. I have updated the breadcrumb to match the call to action link styles when there is no background box

DamienBurkeWN reviewed (approved):
👍
